### PR TITLE
Fixes #18998: Add tests for Rudder client

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/18998

### CODE REVIEW

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

File: agent/sources/client/src/config.h [hunk 1]
```diff
@@ -10,6 +10,8 @@
 
 #define PROTOCOL "https"
 
+#define MAX_ID_LEN 36 + 1
```
Comment (peckpeck) [inline agent/sources/client/src/config.h:13]: what does this mean ?

Comment (amousset) [reply agent/sources/client/src/config.h:13]: UUID len + \0, I'll add a comment

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request merged**